pkg/controllermanager/controller/project/project_stale_control.go

```diff
@@ -172,64 +149,63 @@ func (r *projectStaleReconciler) reconcile(ctx context.Context, project *gardenc
 	return r.gardenClient.Delete(ctx, project)
 }
 
-func (r *projectStaleReconciler) projectInUseDueToShoots(namespace string) (bool, error) {
-	shootList, err := r.shootLister.Shoots(namespace).List(labels.Everything())
-	return len(shootList) > 0, err
+func (r *projectStaleReconciler) projectInUseDueToShoots(ctx context.Context, namespace string) (bool, error) {
+	return kutil.IsNamespaceInUse(ctx, r.gardenClient, namespace, gardencorev1beta1.SchemeGroupVersion.WithKind("ShootList"))
 }
 
-func (r *projectStaleReconciler) projectInUseDueToPlants(namespace string) (bool, error) {
-	plantList, err := r.plantLister.Plants(namespace).List(labels.Everything())
-	return len(plantList) > 0, err
+func (r *projectStaleReconciler) projectInUseDueToPlants(ctx context.Context, namespace string) (bool, error) {
+	return kutil.IsNamespaceInUse(ctx, r.gardenClient, namespace, gardencorev1beta1.SchemeGroupVersion.WithKind("PlantList"))
 }
 
-func (r *projectStaleReconciler) projectInUseDueToBackupEntries(namespace string) (bool, error) {
-	backupEntryList, err := r.backupEntryLister.BackupEntries(namespace).List(labels.Everything())
-	return len(backupEntryList) > 0, err
+func (r *projectStaleReconciler) projectInUseDueToBackupEntries(ctx context.Context, namespace string) (bool, error) {
+	return kutil.IsNamespaceInUse(ctx, r.gardenClient, namespace, gardencorev1beta1.SchemeGroupVersion.WithKind("BackupEntryList"))
 }
 
-func (r *projectStaleReconciler) projectInUseDueToSecrets(namespace string) (bool, error) {
-	secretList, err := r.secretLister.Secrets(namespace).List(labels.Everything())
-	if err != nil {
+func (r *projectStaleReconciler) projectInUseDueToSecrets(ctx context.Context, namespace string) (bool, error) {
+	secretList := &corev1.SecretList{}
+	if err := r.gardenClient.List(ctx, secretList, client.InNamespace(namespace)); err != nil {
```

Comment:
Before, the secret lister was restricted to secrets not controller by gardener using [this labelselector](https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/factory.go#L105).
I think we should restrict listing the secrets with `client.MatchingLabels` similarly here, as listing all secrets in a given namespace can be quite the expensive operation for both client and server.

Reply:
Nice finding, thanks, that wasn't too obvious. PTAL :)